Cargo.toml

```
@@ -202,6 +202,7 @@ try-runtime = [ "polkadot-cli/try-runtime" ]
 fast-runtime = [ "polkadot-cli/fast-runtime" ]
 runtime-metrics = [ "polkadot-cli/runtime-metrics" ]
 pyroscope = ["polkadot-cli/pyroscope"]
+jemalloc-stats = []
```

You can make the `tikv-jemallocator` dependency optional and then enable it here (untested, but I *think* it works):

```suggestion
jemalloc-stats = ["tikv-jemallocator"]
```

---

Actually i just realized that making the dependency optional and enabled only by the flag would make it break on linux, where we want it available whether the flag is present or not.

---

What do you think? I'm undoing the change i made for now.

---

> Actually i just realized that making the dependency optional and enabled only by the flag would make it break on linux, where we want it available whether the flag is present or not.

You can add a Linux-only section in `Cargo.toml` to enable it unconditionally on Linux, e.g. I'd suggest this:

```
[dependencies]
tikv-jemallocator = { version = "0.5.0", optional = true }

[target.'cfg(target_os = "linux")'.dependencies]
tikv-jemallocator = "0.5.0"

[features]
force-use-jemalloc = ["dep:tikv-jemallocator"]
```

Then jemalloc will be always enabled on Linux, and on other platforms you can enable it though `force-use-jemalloc` feature.